File: 05_vetores/ex01_maior_menor.py
"""
Exercicio 01
Tema: 05_vetores
"""

def maior(num, tamanho):
    maior = num[0]

    for i in range(1,tamanho):
        if num[i] > maior:
            maior = num[i]
        
    return maior

def menor(num, tamanho):
    menor = num[0]

    for i in range(1, tamanho):
        if num[i] < menor:
            menor = num[i]
    
    return menor


num = [3,7,1,9,4]

highest = maior(num, len(num))
worst = menor(num, len(num))

print(f"Maior = {highest}")
print(f"Menor = {worst}")

File: 05_vetores/ex01_maior_menor.c
/*
Exercicio 01
Tema: 05_vetores
*/

int maior(int * num, int tamanho){
    int maior = num[0];

    for(int i = 1; i < tamanho; i++){
        if(num[i] > maior){
            maior = num[i];
        }
    }

    return maior;
}

int menor(int * num, int tamanho){
    int menor = num[0];

    for(int i = 1; i < tamanho; i++){
        if(num[i] < menor){
            menor = num[i];
        }
    }

    return menor;
}

#include <stdio.h>

int main() {

    int num[] = {3,7,1,9,4};
    int tamanho = sizeof(num) / sizeof(num[0]);

    int highest = maior(num, tamanho);
    int worst = menor(num, tamanho);

    printf("Maior = %d\n", highest);
    printf("Menor = %d\n", worst);

    return 0;
}
